frontend/ui/screensaver.lua

```diff
@@ -147,4 +147,25 @@ function Screensaver:close()
     end
 end
 
+function Screensaver:isKindleSpecialOffers()
+    -- Look at the current blanket modules to see if the SO screensavers are enabled...
+    local lipc = require("liblipclua")
+    if not lipc then
+        logger.warn("could not load liblibclua")
+        return false
+    end
+    local lipc_handle = lipc.init("com.github.koreader.device")
+    if not lipc_handle then
+        logger.warn("could not get lipc handle")
+        return false
```

Comment:
I know very little about Kindles and even less about those special offers, but from the discussion I got the impression that a whitelist would be better than a blacklist. I think the logic at intoScreensaver() is fine but if there is no better way to do it I literally suggest flipping all the true/false values.

```lua
    if not string.find(loaded_blanket_modules, "ad_screensaver") then
        so = false
    end
```

That still seems rather broad, but at least it wouldn't be a cavalier "oh hey, we had a tiny issue with this lipc thingy so let's display the screensaver."

But I burned my hands on merging the last PR. Let's just see what @NiLuJe thinks. :-)